src/librawspeed/common/Common.h

```diff
@@ -130,21 +130,39 @@ inline Endianness getHostEndianness() {
 # define BSWAP64(A) __builtin_bswap64(A)
 #endif
 
-template<typename T> inline T getByteSwapped(const void* data, bool bswap)
+inline short16 getByteSwapped(short16 v) { return BSWAP16(v); }
+inline ushort16 getByteSwapped(ushort16 v) { return BSWAP16(v); }
+inline int32 getByteSwapped(int32 v) { return BSWAP32(v); }
+inline uint32 getByteSwapped(uint32 v) { return BSWAP32(v); }
+inline uint64 getByteSwapped(uint64 v) { return BSWAP64(v); }
+
+// the float/double versions use two memcpy which guarantee strict aliasing
+// and are compiled into the same assembly as the popular union trick.
+inline float getByteSwapped(float f)
+{
+  uint32 i;
+  memcpy(&i, &f, sizeof(i));
+  i = BSWAP32(i);
+  memcpy(&f, &i, sizeof(i));
+  return f;
+}
+inline double getByteSwapped(double d)
+{
+  uint64 i;
+  memcpy(&i, &d, sizeof(i));
+  i = BSWAP64(i);
+  memcpy(&d, &i, sizeof(i));
+  return d;
+}
+
+template <typename T> inline T getByteSwapped(const void* data, bool bswap)
 {
   T ret;
   // all interesting compilers optimize this memcpy into a single move
-  // this is the most effective way to load some bytes without running into alignmen issues
+  // this is the most effective way to load some bytes without running into
+  // alignment or aliasing issues
   memcpy(&ret, data, sizeof(T));
-  if (bswap) {
-    switch(sizeof(T)) {
-    case 1: break;
-    case 2: ret = BSWAP16(ret); break;
-    case 4: ret = BSWAP32(ret); break;
-    case 8: ret = BSWAP64(ret); break;
-    }
-  }
-  return ret;
+  return bswap ? getByteSwapped(ret) : ret;
```

Comment:
> Your full tests just increased the runtime of the tests by factor of nearly 10. This is out of proportion.

I had to pick /some/ count of the samples, it happened to be the current values. Maybe there should be less samples. The actual ~problem~, is that i want more granulated output from `ctest`. Right now i switched back to less granulated output, and it should be **much** faster.

Also, you do know that you can do `ctest -j9`?

> If you had simply merged that pr as is and did your split of Common.h afterwards, it would have cost you the same amount of work but spared me about 5 minutes for the manual rebase.

Sorry, i agreed to assume maintaintership with thought of *me* being able to practice C++, and improve the library, with *everything* else being secondary. So yes, i guess this is me being somewhat discouraging from active contribution, but i also see no alternatives. Either it works for me, or it does not... Just putting all the cards on the table here.

TLDR: contributing back is important, even though the license does not require it. But *I* need to be motivated to review&merge things.